theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,263 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDescriptor(COp):
+
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('dtype',)
+    params_type = ParamsType(dtype=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        return [os.path.dirname(__file__), config.dnn.include_path]
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        return [config.dnn.library_path]
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, dtype=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+
+        assert cudnn.cudnnDataType_t.has_alias(dtype)
+        self.dtype = dtype
+
+    def make_node(self, dimensions):
```

Comment:
I've used it as a Theano variable, otherwise the grid shape would have to passed by the user, which would repeat the need to pass information on the number of images and channels. Furthermore, the grid dimensions tuple is setup according to the shape of the input image's shape, and the image might be a Theano variable as well.